server/app/assets/javascripts/admin_programs.ts

```diff
@@ -24,88 +24,124 @@ class AdminPrograms {
+      const requiredIndicators = applicationStepOneDiv?.querySelectorAll('span')
+      requiredIndicators?.forEach((indicator) => {
+        if (commonIntakeCheckbox.checked) {
           indicator.classList.add('hidden')
-        })
-      }
+        } else {
+          indicator.classList.remove('hidden')
+        }
+      })
     })
   }
 
-  static maybeDisableApplicationSteps(
-    applicationStepFields: NodeListOf<Element>,
-    commonIntakeCheckbox: HTMLInputElement,
+  /**
+   * Updates the disabled state for multiple text fields matching the provided selector.
+   *
+   * @param fieldSelectors - CSS selector string to identify the text fields to update
+   * @param shouldDisable - Boolean indicating whether to disable (true) or enable (false) the fields
+   */
+  static updateTextFieldSelectorsDisabledState(
+    fieldSelectors: string,
+    shouldDisable: boolean,
   ) {
-    applicationStepFields.forEach((step) => {
-      const applicationStepField = step as HTMLInputElement
-      this.maybeDisableField(applicationStepField, commonIntakeCheckbox.checked)
+    const textFields = document.querySelectorAll(fieldSelectors)
+    textFields.forEach((field) => {
+      const fieldElement = field as HTMLInputElement
+      this.updateTextFieldElementDisabledState(fieldElement, shouldDisable)
     })
   }
 
-  static maybeDisableField(field: HTMLInputElement, shouldDisable: boolean) {
+  /**
+   * Updates the disabled state for a single text field element.
+   *
+   * @param fieldElement - The HTML input element to update
+   * @param shouldDisable - Boolean indicating whether to disable (true) or enable (false) the field
+   */
+  static updateTextFieldElementDisabledState(
+    fieldElement: HTMLInputElement,
+    shouldDisable: boolean,
+  ) {
     if (shouldDisable) {
-      field.disabled = true
-      field.classList.add(
+      fieldElement.disabled = true
+      fieldElement.classList.add(
         this.DISABLED_TEXT_CLASS,
         this.DISABLED_BACKGROUND_CLASS,
       )
     } else {
-      field.disabled = false
+      fieldElement.disabled = false
     }
   }
 
+  /**
+   * Updates the disabled state for USWDS checkboxes matching the provided selector.
+   * When disabling checkboxes, also unchecks them to prevent submitting their values.
+   *
+   * @param fieldSelectors - CSS selector string to identify the checkboxes to update
+   * @param shouldDisable - Boolean indicating whether to disable (true) or enable (false) the checkboxes
+   */
+  static updateUSWDSCheckboxesDisabledState(
+    fieldSelectors: string,
+    shouldDisable: boolean,
+  ) {
+    const checkboxes = document.querySelectorAll(fieldSelectors)
+    checkboxes.forEach((checkbox) => {
+      const checkboxElement = checkbox as HTMLInputElement
+      if (shouldDisable) {
+        checkboxElement.disabled = true
+        checkboxElement.checked = false
+      } else {
+        checkboxElement.disabled = false
+      }
+    })
+  }
```

Comment:
IMO is better to keep the `shouldDisable` boolean than separate on two methods for these ones because it's easier to see what is the enabled and disabled state.
However, I don't feel strongly so can be changed to `enableUSWDSCheckboxes` and `disableUSWDSCheckboxes` (similarly the other methods)